### PR TITLE
feat(solana): basic setup

### PR DESCRIPTION
**Pull request by PeterBenc**
## Description
This PR tries to achieve the following
 - do the most basic setup of solana in the suite UI, just so that solana is there
   - do that without paying attention to anything that could/should be resolved in standalone PR and rather ignore it or mock for now
   - so that checks pass and we can build on top of this in parallel
 - add TODO comments (marked with `vl` for vacuumlabs) in places which may need more attention
    - as we were familiarizing ourselves with the codebase, we found it handy to mark these places so we can look more into them in later separate PRs.
    - The plan is to remove all of these before merging anything into develop.

### CODE REVIEW

File: packages/components/src/components/assets/CoinLogo/coins.ts [hunk 1]
```diff
@@ -20,4 +20,5 @@ export const COINS = {
     vtc: require('../../../images/coins/vtc.svg'),
     xrp: require('../../../images/coins/xrp.svg'),
     zec: require('../../../images/coins/zec.svg'),
+    sol: require('../../../images/coins/ada.svg'), // TODO(vl): add icons
```
Comment (PeterBenc) [inline packages/components/src/components/assets/CoinLogo/coins.ts:23]: I am aware this is not the right way to add a new icon, as you can see we use ada icon for solana for now, expect this to be resolved in separate PR.

File: suite-common/wallet-config/src/networksConfig.ts [hunk 2]
```diff
@@ -417,6 +417,26 @@ export const networks = {
...
+        // features: ['tokens', 'staking'],
+        explorer: {
+            tx: '', // TODO(vl): add explorer links
+            account: '',
+            address: '',
+        },
+        support: {
+            [DeviceModelInternal.T2T1]: '0.0.0', // TODO(vl): always support, update this once we have version
+        },
+        customBackends: [''], // TODO(vl): investigate what this affects
```
Comment (PeterBenc) [inline suite-common/wallet-config/src/networksConfig.ts:435]: Expect `customBackends` to be populated in the following PR

File: suite-common/wallet-core/src/discovery/discoveryThunks.ts [hunk 3]
```diff
@@ -397,13 +397,43 @@ export const startDiscoveryThunk = createThunk(
...
         // prepare bundle of accounts to discover, exclude unsupported account types
-        const bundle = await dispatch(
+        const bundleWithSol = await dispatch(
             getBundleThunk({
                 discovery: { ...discovery, availableCardanoDerivations },
                 device,
             }),
         ).unwrap();
 
+        const solanaDiscoveryItem = bundleWithSol.find(
```
Comment (PeterBenc) [inline suite-common/wallet-core/src/discovery/discoveryThunks.ts:409]: The idea here is to not fetch solana account data from connect, thus bypassing calls for descriptor and for account info. This whole commit will be reverted in the following PR.

**Comment by PeterBenc** [comment]
Also please if we havent followed your repo conventions in any way in this PR or in others, let us know :)

File: suite-native/assets/src/assetsSelectors.ts
Comment (matejkriz) [inline suite-native/assets/src/assetsSelectors.ts:1]: I think the suite-native support is out of scope, but would be nice to double check that.

File: suite-common/wallet-utils/src/accountUtils.ts
Comment (matejkriz) [inline suite-common/wallet-utils/src/accountUtils.ts:1]: Woud be nice to add SOL to [`getTitleForNetwork`](https://github.com/trezor/trezor-suite/pull/9197/files#diff-9beba6400adbb5f2e7c100af7c8af777831a104add04c354f2d2d79ce114d4adR117) so that the Solana account has a name and not just "unknown"

File: suite-common/wallet-core/src/discovery/discoveryThunks.ts (hunk 3 above)
Comment (matejkriz) [reply suite-common/wallet-core/src/discovery/discoveryThunks.ts:409]: would be nice to wrap it in code comment mentioning that it will be reverted, but it's just a nitpick, it has to be reverted to make it really work anyway...

**Review by matejkriz — APPROVED** [review]
Looks good so far, not much for review and that's good. I just think the suite-native is out of scope and there doesn't have to be any changes in that folder now. Just in case any changes for web/desktop break the native mobile app...

File: suite-common/wallet-utils/src/accountUtils.ts
Comment (PeterBenc) [reply suite-common/wallet-utils/src/accountUtils.ts:1]: If its ok, I ve already implemented it in another PR, so you will see that soon.

File: suite-native/assets/src/assetsSelectors.ts
Comment (PeterBenc) [reply suite-native/assets/src/assetsSelectors.ts:1]: I think so too, let me double-check that, I will let you if it isn't out of scope.
Meanwhile I removed all I could from the `native` folder, I had to leave some because ts requires it since `solana` is part of `NetworkType`

File: suite-common/wallet-core/src/discovery/discoveryThunks.ts (hunk 3 above)
Comment (PeterBenc) [reply suite-common/wallet-core/src/discovery/discoveryThunks.ts:409]: https://github.com/trezor/trezor-suite/pull/9197/commits/cc7c085930464ff2fd5274a5aba9d7fc7e2d2096

Comment (matejkriz) [reply suite-common/wallet-core/src/discovery/discoveryThunks.ts:409]: thanks

File: suite-common/wallet-utils/src/accountUtils.ts
Comment (matejkriz) [reply suite-common/wallet-utils/src/accountUtils.ts:1]: sure, no problem

File: suite-native/assets/src/assetsSelectors.ts
Comment (matejkriz) [reply suite-native/assets/src/assetsSelectors.ts:1]: ok, it's same with ripple, that is also not yet supported by mobile app 👌 

We'll probably need to extend this to the array (containing ['sol', 'xrp', 'txrp']) later: https://github.com/trezor/trezor-suite/blob/develop/suite-native/config/src/enabledNetworks.ts#L26
